asm/f64/stubs_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package f64
+
+import (
+	"math"
+	"runtime"
+	"testing"
+)
+
+var (
+	nan = math.NaN()
+	inf = math.Inf(1)
+)
+
+func same(a, b float64) bool {
+	return !(a != b && !math.IsNaN(a) && !math.IsNaN(b) || (math.IsNaN(a) != math.IsNaN(b)))
+}
+
+func TestAdd(t *testing.T) {
+	for j, v := range []struct {
+		dst, src, expect []float64
+	}{
+		{[]float64{1}, []float64{0}, []float64{1}},
+		{[]float64{1, 2, 3}, []float64{1}, []float64{2, 2, 3}},
+		{[]float64{}, []float64{}, []float64{}},
+		{[]float64{1}, []float64{nan}, []float64{nan}},
+		{[]float64{8, 8, 8, 8, 8},
+			[]float64{2, 4, nan, 8, 9},
+			[]float64{10, 12, nan, 16, 17}},
+		{[]float64{0, 1, 2, 3, 4},
+			[]float64{-inf, 4, nan, 8, 9},
+			[]float64{-inf, 5, nan, 11, 13}},
+	} {
+		Add(v.dst, v.src)
```

Comment:
If I do something like this

```
		v.dst = v.dst[1 : len(v.dst)-1]
		Add(dst, v.src)
```

after adding leading and trailing elements above, I get bad things.

```
unexpected fault address 0x0
fatal error: fault
[signal 0xb code=0x80 addr=0x0 pc=0x482ac2]

goroutine 5 [running]:
runtime.throw(0x51e86a, 0x5)
	/home/daniel/Development/go/src/runtime/panic.go:566 +0x8b fp=0xc820031bd0 sp=0xc820031bb8
runtime.sigpanic()
	/home/daniel/Development/go/src/runtime/sigpanic_unix.go:27 +0x27c fp=0xc820031c20 sp=0xc820031bd0
github.com/gonum/internal/asm/f64.Add(0xc820014848, 0x5, 0x6, 0xc820010540, 0x5, 0x5, 0x4, 0x1, 0x1, 0x0, ...)
	/home/daniel/Development/src/github.com/gonum/internal/asm/f64/add_amd64.s:25 +0x32 fp=0xc820031c28 sp=0xc820031c20
github.com/gonum/internal/asm/f64.TestAdd(0xc820078180)
	/home/daniel/Development/src/github.com/gonum/internal/asm/f64/stubs_test.go:37 +0x8dc fp=0xc820031f90 sp=0xc820031c28
testing.tRunner(0xc820078180, 0x535568)
	/home/daniel/Development/go/src/testing/testing.go:547 +0x7a fp=0xc820031fb0 sp=0xc820031f90
runtime.goexit()
	/home/daniel/Development/go/src/runtime/asm_amd64.s:2013 +0x1 fp=0xc820031fb8 sp=0xc820031fb0
created by testing.(*T).Run
	/home/daniel/Development/go/src/testing/testing.go:583 +0x2d9
```